### PR TITLE
Osx travis and AppVeyor

### PR DESCRIPTION
**Pull request by Nirad**
* Fix for OSX build
* Cmake osx toolchain and sphereversion on OS X version
* travis osx auto build script
* Travis and Appveyor rewrite

### CODE REVIEW

**Comment by Nirad** [comment]
i will add ftp stuff in another commit

File: .travis.yml [hunk 1]
```diff
@@ -1,45 +1,44 @@
...
-    build_command_prepend: cmake -G '$CMAKE_GEN' -DCMAKE_BUILD_TYPE="Nightly" -DCMAKE_TOOLCHAIN_FILE='$CMAKE_TCH_64' ./src
-    build_command: make -j4
-      
-script:
-  - if [ $COVERITY_SCAN_BRANCH != 1 ]; then travis_terminate 0; fi
-
-after_script:
-  - echo "Printing Coverity log..."
-  - cat cov-int/scm_log.txt
+matrix:
```
Comment (cbnolok) [inline .travis.yml:2]: Are those changes to .travis.yml necessary, considering that we use Travis-CI only for Coverity, and that it's already working just fine?

File: appveyor.yml [hunk 2]
```diff
@@ -131,10 +135,9 @@ for:
 
       # if we are testing a pull request, building only a version will be sufficient, so stop this script
       - |-
-        if [ -n "${APPVEYOR_PULL_REQUEST_NUMBER}" ]; then
+        if [ "${APPVEYOR_PULL_REQUEST_NUMBER}" ]; then
```
Comment (cbnolok) [inline appveyor.yml:138]: Is there a particular reason why -n was removed?

File: src/cmake/toolchains/OSX-64.cmake [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	# _EXCEPTIONS_DEBUG: Enable advanced exceptions catching. Consumes some more resources, but is very useful for debug
+	#   on a running environment. Also it makes sphere more stable since exceptions are local.
+
+	#-- Set per-build define macros.
+
+	IF (TARGET spheresvr_release)
+		TARGET_COMPILE_DEFINITIONS ( spheresvr_release	PUBLIC NDEBUG )
+	ENDIF (TARGET spheresvr_release)
+	IF (TARGET spheresvr_nightly)
+		TARGET_COMPILE_DEFINITIONS ( spheresvr_nightly	PUBLIC NDEBUG THREAD_TRACK_CALLSTACK _NIGHTLYBUILD )
```
Comment (cbnolok) [inline src/cmake/toolchains/OSX-64.cmake:89]: Are those defines working, given the need to add some of them via add_definitions?

File: src/common/sphere_library/smutex.cpp [hunk 4]
```diff
@@ -9,7 +9,9 @@ SimpleMutex::SimpleMutex()
 	InitializeCriticalSectionAndSpinCount(&m_criticalSection, 0x80000020);
 #else
 	pthread_mutexattr_init(&m_criticalSectionAttr);
+#ifndef __APPLE__
```
Comment (cbnolok) [inline src/common/sphere_library/smutex.cpp:12]: Have you tried one of those solutions to keep the _settype call?
https://stackoverflow.com/questions/18375527/why-is-gnu-source-macro-required-for-pthread-mutexattr-settype-while-it-is-in

File: src/common/CException.cpp [hunk 5]
```diff
@@ -308,6 +308,7 @@ void _cdecl Signal_Break(int sig = 0)		// signal handler attached when using sec
 
 void _cdecl Signal_Illegal_Instruction(int sig = 0)
 {
+#ifndef __APPLE__
```
Comment (cbnolok) [inline src/common/CException.cpp:311]: Can i ask why this security measure was disabled?

File: appveyor.yml (hunk 2 above)
Comment (Nirad) [reply appveyor.yml:138]: no

File: .travis.yml (hunk 1 above)
Comment (Nirad) [reply .travis.yml:2]: oups i forget to re add Coverity stuff. i have add build for windows and OS X for travis i will re add this for Coverity.

File: src/common/CException.cpp (hunk 5 above)
Comment (Nirad) [reply src/common/CException.cpp:311]: yeah, i think cmake toolchain is no right.

File: .travis.yml (hunk 1 above)
Comment (cbnolok) [reply .travis.yml:2]: I mean, in my opinion we can drop those changes to travis.yml since we are using appveyor for automatic deployment

**Comment by cbnolok** [comment]
Any luck with this? It's an interesting addition!

**Comment by Nirad** [comment]
> Any luck with this? It's an interesting addition!

I will get back in this in a couple of days.

**Comment by youhide** [comment]
+1

**Pull request closed without merging**